### DISCUSSION TITLE
Questions about "training_init_model" and "training_reuse_*"

### DISCUSSION THREAD

**Discussion by roger13231** [deepmodeling/dpgen#1922] [Q&A]
Dear developer,

My question is similar to #1485 that I have a large original dataset, but the new dataset is small. In #1485, Dr. Zeng replied that we can use "training_reuse_old_ratio". I have addtional question.

1. For the meaning of "training_init_model". In the dpgen document, the function of "training_init_model" is to keep the model parameters. Do the model parameters mean the trained results, such as learned weights and biases of the neural networks in the DP model, the descriptor/embedding network, and the fitting network? In other words, does `training_init_model: true` mean that the model in the current iteration is initialized from the trained model of the previous iteration, instead of being trained from random initialization?

2. For the "training_reuse_*" setting. Should I use these parameters together with "training_init_model" for my new iteration? 
such as "training_init_model": true,
     "training_reuse_old_ratio": 0.3,
     "training_reuse_iter": 135,
     "training_reuse_numb_steps": 400000,
     "training_reuse_start_lr": 0.001
And what is the recommendation for "training_reuse_old_ratio", 0.2, 0.3, or some other ratio?

**Comment by dosubot** [comment]
Hi @roger13231! I'm [Dosu](https://go.dosu.dev/dosubot) and I'm helping the DeepModeling team.

**1. Regarding `training_init_model`:**

Yes, your understanding is correct. When `training_init_model: true`, the model in the current iteration is initialized from the trained model of the previous iteration — preserving all learned weights, biases, descriptor/embedding network, and fitting network parameters — rather than being trained from random initialization [[1]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/arginfo.py#L101). When `iter == 0`, it initializes from `training_iter0_model_path` instead [[2]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/arginfo.py#L149).

**2. Regarding `training_reuse_*` parameters:**

Yes, these should be used **together** with `training_init_model: true`. The system works as follows [[3]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/run.py#L507-L513):

- `training_reuse_iter` — the iteration index at which model reuse mode activates [[4]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/arginfo.py#L105)
- `training_reuse_old_ratio` — the probability of sampling old data during training. A value of `0.3` means 30% old data, 70% new data [[5]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/arginfo.py#L107-L117)
- `training_reuse_numb_steps` — overrides the number of training steps during reuse iterations [[6]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/arginfo.py#L118)
- `training_reuse_start_lr` — overrides the starting learning rate for reuse iterations [[7]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/run.py#L550-L551)

Your example configuration looks reasonable:

```json
"training_init_model": true,
"training_reuse_old_ratio": 0.3,
"training_reuse_iter": 135,
"training_reuse_numb_steps": 400000,
"training_reuse_start_lr": 0.001
```

**Regarding the recommended `training_reuse_old_ratio`:**

For your scenario (large original dataset + small new dataset), a value in the range of **0.3–0.5** is reasonable to boost the probability of new data being selected during training [[8]](https://github.com/deepmodeling/dpgen/issues/1485). Note that:

- A lower value (e.g., `0.3`) gives **more** weight to new data (70% new, 30% old)
- You can also use `"auto"` which applies an automatic ratio calculation (default 10:1 new-to-old), or `"auto:f"` where `f` is a custom new-to-old ratio factor [[5]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/dpgen/generator/arginfo.py#L107-L117)

A test case in the codebase uses `training_reuse_old_ratio=0.8` with `training_reuse_start_lr=1e-4` [[9]](https://github.com/deepmodeling/dpgen/blob/0a08fd38e1a2403df8e600c3d5be5f93e130e802/tests/generator/test_make_train.py#L324-L366), but that's for a scenario where you want to keep most old data. Since your new dataset is small relative to the original, using `0.3` as you proposed should help the model learn from the new configurations effectively.



*To reply, just mention [@dosu](https://go.dosu.dev/dosubot?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=reply-with-mention&utm_term=deepmodeling%2Fdpgen).*

---

Docs are dead. Just use [Dosu](https://dosu.dev?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-tagline&utm_term=deepmodeling%2Fdpgen).

[![Leave Feedback](https://img.shields.io/badge/Leave%20Feedback-555555?style=flat)](https://app.dosu.dev/response-feedback?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-feedback&utm_term=deepmodeling%2Fdpgen&message_id=d7150b2a-a8b8-441a-b169-712815538437) [![Ask Dosu about dpgen](https://img.shields.io/badge/Ask%20Dosu%20about%20dpgen-2f7b3f?style=flat&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CPHN2ZyB3aWR0aD0iODYiIGhlaWdodD0iODkiIHZpZXdCb3g9IjAgMCA4NiA4OSIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48cGF0aCBkPSJNNS4yOTIzNiAxMi43OTI4TDE3Ljc1OTMgNi42ODE4OFY3Mi41NjY3TDUuMjkyMzYgODQuMDYxOFYxMi43OTI4WiIgZmlsbD0iI0I0QkI5MSIvPjxwYXRoIGQ9Ik0xOC4yNTc1IDczLjExOTZMNTkuMTMyOSA3Mi43NDhMNTEuNzAxMSA4Mi40MDk1TDI5LjAzMzggODYuMjkxTDYuMjM5NjIgODUuMTU1NEwxOC4yNTc1IDczLjExOTZaIiBmaWxsPSIjNzc4NTYxIi8%2BPHBhdGggZD0iTTE3LjQ5MTYgMy43MzYzM0wzLjU4NTU3IDEyLjcwOTlWODMuNTc5MkMzLjU4NTU3IDg0Ljc1NDIgNC45ODU2MyA4NS4zNjUyIDUuODQ3MDUgODQuNTY2TDE5LjYyOTYgNzEuNzgwMSIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSI2LjQyODQ0IiBzdHJva2UtbGluZWNhcD0icm91bmQiLz48bWFzayBpZD0iZG9zdS1kLWN1dG91dCIgZmlsbD0id2hpdGUiPjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNNDAuNzA0IDAuNTE4MDY2SDE3LjA0MzlWNzYuMjIyMUg0MC43MDRINDIuNTgwNUg0Ny44MDEzQzY4LjcwNjQgNzYuMjIyMSA4NS42NTMzIDU5LjI3NTIgODUuNjUzMyAzOC4zNzAxQzg1LjY1MzMgMTcuNDY1IDY4LjcwNjMgMC41MTgwNjYgNDcuODAxMyAwLjUxODA2Nkg0Mi41ODA1SDQwLjcwNFoiLz48L21hc2s%2BPHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik00MC43MDQgMC41MTgwNjZIMTcuMDQzOVY3Ni4yMjIxSDQwLjcwNEg0Mi41ODA1SDQ3LjgwMTNDNjguNzA2NCA3Ni4yMjIxIDg1LjY1MzMgNTkuMjc1MiA4NS42NTMzIDM4LjM3MDFDODUuNjUzMyAxNy40NjUgNjguNzA2MyAwLjUxODA2NiA0Ny44MDEzIDAuNTE4MDY2SDQyLjU4MDVINDAuNzA0WiIgZmlsbD0iI0YzRjZGMSIvPjxwYXRoIGQ9Ik0xNy4wNDM5IDAuNTE4MDY2Vi02LjU3OTE5SDkuOTQ2NjlWMC41MTgwNjZIMTcuMDQzOVpNMTcuMDQzOSA3Ni4yMjIxSDkuOTQ2NjlWODMuMzE5NEgxNy4wNDM5Vjc2LjIyMjFaTTE3LjA0MzkgNy42MTUzMkg0MC43MDRWLTYuNTc5MTlIMTcuMDQzOVY3LjYxNTMyWk0yNC4xNDEyIDc2LjIyMjFWMC41MTgwNjZIOS45NDY2OVY3Ni4yMjIxSDI0LjE0MTJaTTQwLjcwNCA2OS4xMjQ5SDE3LjA0MzlWODMuMzE5NEg0MC43MDRWNjkuMTI0OVpNNDIuNTgwNSA2OS4xMjQ5SDQwLjcwNFY4My4zMTk0SDQyLjU4MDVWNjkuMTI0OVpNNDcuODAxMyA2OS4xMjQ5SDQyLjU4MDVWODMuMzE5NEg0Ny44MDEzVjY5LjEyNDlaTTc4LjU1NiAzOC4zNzAxQzc4LjU1NiA1NS4zNTU1IDY0Ljc4NjcgNjkuMTI0OSA0Ny44MDEzIDY5LjEyNDlWODMuMzE5NEM3Mi42MjYxIDgzLjMxOTQgOTIuNzUwNSA2My4xOTQ5IDkyLjc1MDUgMzguMzcwMUg3OC41NTZaTTQ3LjgwMTMgNy42MTUzMkM2NC43ODY2IDcuNjE1MzIgNzguNTU2IDIxLjM4NDcgNzguNTU2IDM4LjM3MDFIOTIuNzUwNUM5Mi43NTA1IDEzLjU0NTMgNzIuNjI2IC02LjU3OTE5IDQ3LjgwMTMgLTYuNTc5MTlWNy42MTUzMlpNNDIuNTgwNSA3LjYxNTMySDQ3LjgwMTNWLTYuNTc5MTlINDIuNTgwNVY3LjYxNTMyWk00MC43MDQgNy42MTUzMkg0Mi41ODA1Vi02LjU3OTE5SDQwLjcwNFY3LjYxNTMyWiIgZmlsbD0iYmxhY2siIG1hc2s9InVybCgjZG9zdS1kLWN1dG91dCkiLz48cGF0aCBkPSJNNjguOTIxNSAzNi4wMTM1QzY4LjkyMTUgMzYuMDEzNSA2NS43MzY5IDQ5LjQ3MzggNTEuNDIzMSA0OS40NzM4QzM3LjEwOTMgNDkuNDczOCAzMi41Nzg3IDM3LjM1OTYgMzIuNTc4NyAzNi4wMTM1IiBzdHJva2U9ImJsYWNrIiBzdHJva2Utd2lkdGg9IjcuNjkxNjEiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCIvPjxwYXRoIGQ9Ik0wLjM0ODYzMyA4NS40OTQ2QzAuMzQ4NjMzIDg1LjQ5NDYgMjkuNDg1NiA4NS44MzA5IDM0LjgwOSA4NS42OThDNDQuODMzNyA4NS40NDc3IDUxLjI4NzIgODQuNDAyIDU3LjUyNjkgNzguOTcyNEM2Mi44MTI5IDc0LjM3MjcgNzUuMTM0MiA1OS42ODM2IDc1LjEzNDIgNTkuNjgzNiIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSI2LjE2NDgyIi8%2BPC9zdmc%2B)](https://github.dosu.com/deepmodeling/dpgen?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-ask-repo&utm_term=deepmodeling%2Fdpgen) [![Share Dosu with your team](https://img.shields.io/badge/Share%20Dosu%20with%20your%20team-1f6feb?style=flat)](https://app.dosu.dev/signup?referrer=openSource&source=github-footer&utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-share-team&utm_term=deepmodeling%2Fdpgen)

[a comment by an automated account is not shown]
